app/get-page.js

```diff
@@ -42,33 +47,60 @@ async function getPageContents (url, requestMethod) {
   return {
     contents: content.toString(),
     status: {
-      'url': url,
-      'content_type': response.headers['content-type'],
-      'content_length': contentLength,
-      'http_code': response.statusCode,
-    }
+      url: url,
+      content_type: response.headers['content-type'],
+      content_length: contentLength,
+      http_code: response.statusCode,
+    },
   }
 }
 
 async function request (url, requestMethod) {
   try {
-    const options = {
-      'method': requestMethod,
-      'encoding': null,
-      'headers': {'user-agent': process.env.USER_AGENT || DEFAULT_USER_AGENT}
+    let options, response
+    const dat = await redisGet(url + requestMethod)
+    if (dat) {
+      const body = JSON.parse(dat).body
+      const e = JSON.parse(dat).etag
+
+      options = {
+        method: requestMethod,
+        encoding: null,
+        headers: {
+          'user-agent': process.env.USER_AGENT || DEFAULT_USER_AGENT,
+          'if-none-match': e,
+        },
+      }
+      response = await got(url, options)
+      response.body = Buffer.from(body)
+      return processContent(response)
+    }
+    options = {
+      method: requestMethod,
+      encoding: null,
+      headers: {'user-agent': process.env.USER_AGENT || DEFAULT_USER_AGENT},
```

Comment:
Here, `options` can be a `const` as the original code,  just the `if-none-match` header need to be set
```js
options.headers['if-none-match'] = e;
```

Reply:
Hello @gnuns ! Thanks for your suggestions. I have changed options obj from let to const.